jazzmin/static/jazzmin/js/change_form.js

```diff
@@ -115,7 +115,10 @@ $(document).ready(function () {
 
     // Ensure we preserve the tab the user was on using the url hash, even on page reload
 
-    if ($tabs.length) {handleTabs($tabs);}
-    else if ($carousel.length) {handleCarousel($carousel);}
-    else if ($collapsible.length) {handleCollapsible($collapsible);}
+    if ($tabs.length) { handleTabs($tabs); }
+    else if ($carousel.length) { handleCarousel($carousel); }
+    else if ($collapsible.length) { handleCollapsible($collapsible); }
+
+    // Apply select2 to any select boxes that dont yet have it
+    $('select').not('.select2-hidden-accessible').select2({ dropdownAutoWidth: true });
```

Comment:
@farridav while testing, I've seen that this is adding a text input on m2m filters, as they also have a hidden select2

<img width="743" alt="Screenshot 2020-10-12 at 16 55 02" src="https://user-images.githubusercontent.com/22962693/95766837-c2148e00-0cab-11eb-96c0-7928199c4db2.png">